Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
```

Comment:
Neither `ThreadConnectionPool` nor `storage` are static. I don't seem to see how the `storage` dictionary is going to ever contain more than one connection. Also, seems like threads can only create one connection, mapped to their thread id, and access that connection only. Once a connection is used by a thread, how would other threads use it if the thread id does not match?